omero/developers/json-api.txt

```diff
@@ -0,0 +1,1329 @@
+
+The major version is included in the URL such as ``/v0/`` whereas the full
+version number can be found in the header:
+
+::
+
+    X-OMERO-ApiVersion: 0.0
+
+
+JSON format
+^^^^^^^^^^^
+
+The JSON objects generated by `omero-marshal <https://github.com/openmicroscopy/omero-marshal>`_
+are defined by the :model_doc:`OME-Model <ome-xml/>`. The OMERO model closely follows the
+OME schema but is not identical. In the cases where OMERO-specific fields are included, these
+will be prefixed by ``omero:``. For example, ``omero:details`` specifies the owner, group and
+permissions of each object in OMERO.
+JSON objects also include an ``@id`` of the object in the OMERO database and a ``@type`` that specifies
+the OME Schema used to generate it such as ``http://www.openmicroscopy.org/Schemas/OME/2016-06#Project``.
+
+All the fields of the OMERO model object will be included in the JSON except those that are ``null``,
+which will be omitted.
+Where supported, modifying the JSON object and saving this back to OMERO will update the object accordingly.
+
+
+URLs in JSON
+^^^^^^^^^^^^
+
+URLs are included in JSON objects using keys with the ``url:`` prefix. URLs are added for
+related objects to facilitate exploration of the API in a browser. You may find that a
+JSON formatting plugin for your browser improves both the presentation and
+navigation of JSON data.
+
+
+Pagination
+^^^^^^^^^^
+
+Requests that return a list of items will be paginated, showing
+a ``limit`` of the first 200 objects by default.
+Pagination can be specified using the ``limit`` and ``offset`` query
+parameters:
+
+::
+
+    # List the first 100 Projects (offset=0 by default)
+    <server>/api/v0/m/projects/?limit=100
+
+    # List the next 100 Projects
+    <server>/api/v0/m/projects/?limit=100&offset=100
+
+
+Pagination details will be returned in a ``meta`` json object, including
+the ``totalCount`` of objects for that query, the current ``offset`` and
+``limit`` as well as the ``maxLimit`` that you can use.
+
+::
+
+    "meta": {
+        "totalCount": 13240,
+        "maxLimit": 500,
+        "limit": 200,
+        "offset": 0
+    },
+
+Sysadmins can configure the default ``limit`` and ``maxLimit`` settings
+for their server, for example:
+
+::
+
+    $ bin/omero config set omero.web.api.limit 100
+    $ bin/omero config set omero.web.api.max_limit 300
+
+The ``maxLimit`` setting prevents API consumers from requesting very
+large amounts of data by limiting the number of top-level objects that
+are loaded.
+
+
+Loading of linked objects
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In most cases the API loads only the requested objects along with
+their ``omero:details``. For example, ``/api/v0/m/projects/`` loads
+Projects but does not also load their child Datasets.
+However, it is sometimes useful to load a number of closely related objects.
+For example, loading Images also loads their Pixels data (but not Channels)
+and loading Wells also loads WellSamples and Images (but not Pixels).
```

Comment:
The term "fields" is used later on in the page.
So might be good to put WellSamples (fields) here too